crates/bevy_ui/src/render/debug_overlay.rs

```diff
@@ -0,0 +1,98 @@
+use bevy_asset::AssetId;
+use bevy_color::Hsla;
+use bevy_ecs::entity::Entity;
+use bevy_ecs::system::Commands;
+use bevy_ecs::system::Query;
+use bevy_ecs::system::Res;
+use bevy_ecs::system::ResMut;
+use bevy_ecs::system::Resource;
+use bevy_math::Rect;
+use bevy_math::Vec2;
+use bevy_render::sync_world::RenderEntity;
+use bevy_render::sync_world::TemporaryRenderEntity;
+use bevy_render::Extract;
+use bevy_sprite::BorderRect;
+use bevy_transform::components::GlobalTransform;
+
+use crate::ComputedNode;
+use crate::DefaultUiCamera;
+use crate::TargetCamera;
+
+use super::ExtractedUiItem;
+use super::ExtractedUiNode;
+use super::ExtractedUiNodes;
+use super::NodeType;
+
+#[derive(Resource)]
+pub struct UiDebugOptions {
+    pub enabled: bool,
+    pub line_width: f32,
+}
+
+impl Default for UiDebugOptions {
+    fn default() -> Self {
+        Self {
+            enabled: true,
+            line_width: 3.,
+        }
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn extract_debug_overlay(
+    mut commands: Commands,
+    debug_options: Extract<Res<UiDebugOptions>>,
+    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
+    default_ui_camera: Extract<DefaultUiCamera>,
+    uinode_query: Extract<
+        Query<(
+            Entity,
+            &ComputedNode,
+            &GlobalTransform,
+            Option<&TargetCamera>,
+        )>,
+    >,
+    mapping: Extract<Query<RenderEntity>>,
+) {
+    if !debug_options.enabled {
+        return;
+    }
+
+    for (entity, uinode, transform, camera) in &uinode_query {
+        let Some(camera_entity) = camera.map(TargetCamera::entity).or(default_ui_camera.get())
+        else {
+            continue;
+        };
+
+        let Ok(render_camera_entity) = mapping.get(camera_entity) else {
+            continue;
+        };
+
+        extracted_uinodes.uinodes.insert(
+            commands.spawn(TemporaryRenderEntity).id(),
+            ExtractedUiNode {
+                stack_index: uinode.stack_index + 2_147_483_647,
```

Comment:
Does this magic value have a better way of writing it? Maybe a constant?

Reply:
Yeah, this is `i32::MAX`

Reply:
It's just a big number added to the uinode's stack index so the overlay is drawn on top.

Reply:
Using a constant and adding a comment are both good enhancements.